code-samples/calendar/python/create_calendar_event/create_calendar_event/__main__.py

```diff
@@ -0,0 +1,202 @@
+        calendar_id (str): The ID of the calendar to read events from.
+        start_time (int): The start time of the range in Unix timestamp format.
+        end_time (int): The end time of the range in Unix timestamp format.
+
+    Returns:
+        list: A list of calendar events within the specified time range.
+
+    Raises:
+        nylas.models.errors.NylasApiError: If there is an error while fetching calendar events.
+    """
+    try:
+        # Get events from the specified calendar within the time range
+        query_params = ListEventQueryParams(
+            limit=20, calendar_id=calendar_id, start=start_time, end=end_time
+        )
+
+        events, _, _ = client.events.list(
+            identifier=os.environ.get("GRANT_ID"), query_params=query_params
+        )
+
+        return list(events)
+    except NylasApiError as e:
+        print(f"Error while fetching calendar events: {e}")
+
+
+def create_calendar_event(
+    client,
+    title,
+    location,
+    start_time,
+    end_time,
+    description=None,
+    participants=None,
+    calendar_id=None,
+):
+    """
+    Create a new calendar event and save it inside a specific calendar.
+
+    Args:
+        client (nylas.APIClient): The initialized Nylas API client.
+        title (str): The title or name of the event.
+        location (str): The location where the event will take place.
+        start_time (int): The start time of the range in Unix timestamp format.
+        end_time (int): The end time of the range in Unix timestamp format.
+        description (str, optional): A description or additional information about the event.
+        participants (list, optional): List of participant dictionaries with an "email" field.
+        calendar_id (str, optional): The ID of the calendar where the event should be saved.
+
+    Returns:
+        nylas.events.Event: The created event object.
+
+    Raises:
+        nylas.client.errors.NylasError: If there's an issue with the Nylas API.
+
+    Examples:
+        Create a simple event with participants and save it in a specific calendar:
+        >>> participants = [{'email': 'oss1@wiseai.dev'}, {'email': 'oss2@wiseai.dev'}]
+        >>> event = create_calendar_event(client, 'Meeting', 'Conference Room', 1729457999, 1729458000, participants, 'your_calendar_id' )
+
+        Create an event with a description and participants and save it in a specific calendar:
+        Create an event with a description and participants and save it in a specific calendar:
+        >>> participants = [{'email': 'oss1@wiseai.dev'}, {'email': 'oss2@wiseai.dev'}]
+        >>> event = create_calendar_event(client, 'Team Meeting', 'Virtual', 1729457999, 1729458000, 'Discuss project progress', participants, 'your_calendar_id')
+    """
+    try:
+        # create an event for that calendar ID
+        # request_body = CreateEventRequest(title=title, description=description, location=location, when=when, calendar_id=calendar_id)
```

Comment:
There are two ways to create a request body, either using this class, or a dict. The former has an argument `when` of type `nylas.models.events.When` which is not a `dict`, but rather a `Union` type, so not sure how to initialize it.